### ISSUE TITLE
unkdown issue

### ISSUE THREAD

**Issue by mcolominas** [kirschbaum-development/eloquent-power-joins#32]
Hi, I'm here again, and I'm sure you're going to hate me.

I have found a issue, and it was difficult for me to replicate it to give you an example, since I could not see where it came from, here I write the details:

From Code:

```php
for ($i = 0; $i < 2; $i++) {
    Page::joinRelationship('translation')->first();
}
Page::with('translations')->get();
dd(Page::joinRelationship('translation')->toSql());
```

//Output
select `pages`.* from `pages`
________________________________________________________
From Code:

```php
for ($i = 0; $i < 1; $i++) {
    Page::joinRelationship('translation')->first();
}
Page::with('translations')->get();
dd(Page::joinRelationship('translation')->toSql());
```

//Output
select `pages`.* from pages` inner join `page_translations` on `page_translations`.`page_id` = `pages`.`id` and `page_translations`.`locale` = ?
________________________________________________________
From Code:

```php
for ($i = 0; $i < 2; $i++) {
    Page::joinRelationship('translation')->first();
}
//Page::with('translations')->get();
dd(Page::joinRelationship('translation')->toSql());
```

//Output
select `pages`.* from pages` inner join `page_translations` on `page_translations`.`page_id` = `pages`.`id` and `page_translations`.`locale` = ?

Code 2 and 3 are correct, but code 1 is wrong, it is very rare and only appears in certain cases as I have shown you in the examples that I have replicated.

I hope you can know why this happens because I have no idea.

**Comment by luisdalmolin** [comment]
@mcolominas Hi there. Yeah, this seems like a valid issue, thanks for the steps to replicate. I'll take a look at this probably this weekend.

**Comment by luisdalmolin** [comment]
@mcolominas can you try to prefix your calls with `query()` to see what if you still can replicate the issue? e.g.

```php
for ($i = 0; $i < 2; $i++) {
    Page::query()->joinRelationship('translation')->first();
}
Page::query()->with('translations')->get();
dd(Page::query()->joinRelationship('translation')->toSql());
```

**Comment by mcolominas** [comment]
Running the code you have given me, it returns:
select `pages`.* from `pages`

**Comment by mcolominas** [comment]
Also, add a prefix or I do not think that is the problem, what is strange is that when it is executed X times or more the same relation is when it fails.

Test combinations:

```php
//Success
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```
```php
//Fail
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```
```php
//Success
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```
```php
//Fail
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```
```php
//Fail
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```

```php
//Success
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```

```php
//Success
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```

```php
//Fail
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```

```php
//Fail
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```

```php
//Success
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->joinRelationship('translation')->first();
Page::query()->with('translation')->first();
Page::query()->joinRelationship('translation')->first();
dd(Page::query()->joinRelationship('translation')->toSql());
```

**Comment by luisdalmolin** [comment]
@mcolominas This should be fixed in the 2.2.3 version, can you check?
